sumologic/resource_sumologic_metrics_search.go

```diff
@@ -0,0 +1,216 @@
+package sumologic
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceSumologicMetricsSearch() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSumologicMetricsSearchCreate,
+		Read:   resourceSumologicMetricsSearchRead,
+		Update: resourceSumologicMetricsSearchUpdate,
+		Delete: resourceSumologicMetricsSearchDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"title": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"parent_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"log_query": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"desired_quantization_in_secs": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"time_range": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: GetTimeRangeSchema(),
+				},
+			},
+			"properties": {
```

Comment:
I assume this property is a JSON string. I wouldn't expose this property unless absolutely necessary. Given that this is a JSON string, provider will complain about diff if the order of fields changes within JSON (even though nothing changed). This can be a source of frustration for end users. I would take out this field and add it later only if there is an explicit ask for it.